smart ads package (#33)
frontity/api-reference: docs-api/frontity-packages/features-packages/smart-ads.md

---
description: API reference of `@frontity/smart-adserver` package
---

# @frontity/smart-adserver

> Smart Adserver is an ad serving network and they provide an API for displaying the ads which is described [here](https://support.smartadserver.com/s/article/Tagging-guide)

Reviewer: Just a missing `.` at the end of the sentence.

```suggestion
Smart Adserver is an ad serving network and they provide an API for displaying the ads which is described [here](https://support.smartadserver.com/s/article/Tagging-guide).
```

> The `@frontity/smart-adserver` package will load a third-party Smart Adserver library which adds certain properties on the global `window` object. Then you as the developer can then use those properties to make the "ad call" (which are basically API calls to the Smart Adserver).
> 
> In response, the Smart Adserver dynamically loads some code which will modify the DOM to insert the ad in the place that was specified in the API call.

Reviewer: Uhm, this is not completely rght. It's true that `@frontity/smart-adserver` loads a third-party library, but developers wouldn't have to interact with it. The one that uses the SmartAdserver API is the `<SmartAd>` component exposed in `libraries.fills.smartAdserver.SmartAd`.

I would write something like this instead:

```suggestion
The `@frontity/smart-adserver` package will load a third-party Smart Adserver library which adds certain properties on the global `window` object. Then you as the developer can render the exposed `<SmartAd>` component, which will use those properties to make the "ad call" (which are basically API calls to the Smart Adserver).

In response, the Smart Adserver dynamically loads some code which will modify the DOM to insert the ad in the place that the `<SmartAd>` component was rendered.
```

The package has 3 main components:

- _The "Root" component_. It includes the `<Head>` that loads the Smart Adserver library. When the user adds the `@frontity/smart-adserver` to their `frontity.settings.js` file, this library will be loaded automatically.

> - _The `SmartAd` component_. This component is exposed in `libraries.fills.SmartAd`. The users can just use this component directly to display ads by passing it relevant props. The component takes care of calling the Smart Adserver API and injecting the ad into the DOM in the relevant place

Reviewer: The path where the component is exposed is wrong:

```suggestion
- _The `SmartAd` component_. This component is exposed in `libraries.fills.SmartAdserver.SmartAd`. The users can just use this component directly to display ads by passing it relevant props. The component takes care of calling the Smart Adserver API and injecting the ad into the DOM in the relevant place
```

| **`siteId`**   | number | yes      | The `id` of the site as defined in [SmartAds](https://support.smartadserver.com/s/article/Ad-API-reference); represents a website or mobile application; [_more info_](https://support.smartadserver.com/s/article/Setting-up-inventory)                          |
| **`pageId`**   | number | yes      | The `id` of the page as defined in [SmartAds](https://support.smartadserver.com/s/article/Ad-API-reference); represents a website or mobile application; [_more info_](https://support.smartadserver.com/s/article/Setting-up-inventory)                          |
| **`formatId`** | number | yes      | The `id` of the format as defined in [SmartAds](https://support.smartadserver.com/s/article/Ad-API-reference); represents an ad slot on a page (medium rectangle, skyscraper...); [_more info_](https://support.smartadserver.com/s/article/Setting-up-inventory) |
| **`callType`** | string | yes      | The type of the ad call. Possible values: `iframe` or `std`                                                                                                                                                                                                       |

> | **`tagId`**    | string | no       | The `id` of the container that will contain the ad. Default Value: `sas_${formatId}"`                                                                                                                                                                             |
> 
> | **`width`** | number | no | The width of the ad. Used with callType `iframe`. |

Reviewer: The table here is broken, there is an new line character that should be removed (line 91).

```suggestion
| **`tagId`**    | string | no       | The `id` of the container that will contain the ad. Default Value: `sas_${formatId}"`                                                                                                                                                                             |
| **`width`** | number | no | The width of the ad. Used with callType `iframe`. |
```
